Added Impact to the use cases to justify requirements. (#87)
SpencerDawkins/net-collab-rqmts: draft-kwbdgrr-tsvwg-net-collab-rqmts.md

: This is a per-packet metadata requirement.

## Client-to-Network

REQ-CLIENT-DECIDES:
: User/Client indicating to the network to honor the application's
metadata signaling.
: This is a per-flow metadata requirement.

## API

REQ-API-FRAMEWORK:
: API framework to facilitate signaling for applications.
: Signaling to the network ({{client-network}}, {{server-network}})
will need to be facilitated by Application Programming Interfaces
(APIs) for any application to use them. Signaling and retrieval of
the signals may not be performed at a single layer (although not
encouraged). Hence, a framework is required to abstract the underlying
protocol(s) and allow the application(s) to retrieve/send signals
using a single or a set of APIs independent of the channels that
are used to convey the signals. The API framework is required even
if one single channel is used so that any application on a client can
consume the signals.
: There might be many channels to signal the metadata such as
(non-exhaustive list):

  * TCP options {{?RFC9293}}
  * UDP Options {{?I-D.ietf-tsvwg-udp-options}}
  * IPv6 Hop-by-Hop Options ({{Section 4.3 of ?RFC8200}})
  * QUIC CID mapping
  * ICMP messages

> ## System-Considerations

Reviewer: ```suggestion
## System Considerations
```

REQ-PRIVACY-ADDITIONAL:

> : An on-path observer obtains no additional information about the IP

Reviewer: ```suggestion
: An on-path observer obtains (or gleans) no additional information about the IP
```

packet.

REQ-SIGNALING-AVOIDANCE:

> : Leveraging previous experience {{?RFC9049}}, the folliwing is not

Reviewer: ```suggestion
: Leveraging previous experience {{?RFC9049}}, the following is not
```

required to make use of the collaborative signaling:

  * Reveal the application identity.
  * Expose the application cause (or 'reason') to signal metadata.

>   * Reveal server identity.

Reviewer: ```suggestion
  * Reveal the server identity.
```

  * Inspect client-to-server encrypted payload by network elements.

REQ-ISP-SCALE:

> : The metadata other state information that a wireless router has to

Reviewer: ```suggestion
: The metadata and any other related state information that a network element (router) has to
```

> maintain for each additional media flow it handles should be very low

Reviewer: ```suggestion
maintain for each additional flow it handles should be very low
```

or none.

REQ-SIGNAL-VALIDATION:
: The network/OS needs to ensure that the user/client signaling of
priority (if any) does not associate the same priority level with all
traffic types within the same flow, thereby avoiding prioritizing of
all the streams/traffic the same way.

REQ-CLIENT-VALIDATION:
: The network needs to ensure the signal is coming from the same
user/client that is part of the 5-tuple flow.  This is to ensure no
other application influences the priority of another application's

> flow.

Reviewer: ```suggestion
flow from within the same host.
```

1. Audio is more critical than video for many applications and should
be prioritized differently than video.

> The client may indicate this preference to the network by selecting

Reviewer: ```suggestion
The client may indicate this preference to a network by selecting
```

the importance value of audio packets as the highest priority.

    Requirement: REQ-PACKET-PRIORITY, REQ-MEDIA-AV-SEPARATE,
    REQ-PAYLOAD-CLIENT-DECIDES (defined in {{x-req-def}}).

    Impact: With the above requirement met, audio clarity could be maintained

>   better in challenging network and during reactive events ensuring better user experience.

Reviewer: ```suggestion
  better in resource-constrained networks and during reactive events ensuring better user experience.
```

2. The server (or relay) sends the same stream to many receivers,
including the same metadata (especially with media over QUIC).
Different clients have different priorities for different types of traffic. This would result

> in change in priorities for the same type of traffic a single server sends, based on the user/client.

Reviewer: ```suggestion
in change in priorities for the same type of traffic that a single server sends, based on the user/client.
```

    Requirement: REQ-PAYLOAD-CLIENT-DECIDES.

    Impact: With the above requirement met, each client/user preferences are
    prioritized accordingly while maintaining scalability on the server, since

>     the metadata the server sends still remains the same for all the connections.

Reviewer: ```suggestion
    the metadata that the server sends still remains the same for all the connections.
```

3. Video contains partial frames and full frames, which need to be
distinguished so that full frames can be indicated to the network.
The application may, for example, mark all packets of key frames with the
highest priority to indicate that they should not be dropped.

    Requirement: REQ-PACKET-PRIORITY, REQ-MEDIA-KEYFRAME (defined in {{x-req-def}}).

    Impact: Streaming continuity is improved with lesser impact to continuity/quality
    of service by prioritizing full frames during reactive events and in challenging networks.

4. In loss-prone networks or during Reactive Management events, if all packets of an application
flow (UDP 4-tuple) such as live broadcast or on-demand video streaming are treated the same,
it limits the ability to maximize network utilization and use the transiently available bandwidth.
Dropping or delaying of (media) packets randomly is likely to lower network utilization
and application performance.

    Requirement: REQ-PACKET-PRIORITY.

>     Impact: By identifying packets that can be dropped, congestion can be reduced leading

Reviewer: ```suggestion
    Impact: By identifying packets that tolerate being dropped, congestion can be reduced leading
```
